gdc_client/upload/client.py

```diff
@@ -135,62 +135,79 @@ def __init__(self, token, processes, server, part_size,
+
         if not (server.startswith('http://') or server.startswith('https://')):
             server = 'https://' + server
+
         self.server = server
         self.multipart = multipart
         self.upload_id = None
         self.debug = debug
         self.processes = processes
         self.part_size = (max(part_size, MIN_PARTSIZE)/PAGESIZE+1)*PAGESIZE
-        self._metadata = None
+        self._metadata = {}
         self.resume_path = "resume_{}".format(self.manifest_name)
 
-    @property
-    def metadata(self):
-        return self._metadata or self.get_metadata(self.node_id)
+    def metadata(self, field):
+        return self._metadata.get(field) or self.get_metadata(self.node_id)[field]
 
     def get_metadata(self, id):
         '''
         Get file's project_id and filename from graphql
         '''
-        self._metadata = None
-        query = {'query':
-                 """query Files { node (id: "%s") { type }}""" % id}
+
+        # first get the file_type
+        self._metadata = {}
+        query = {'query': 'query Files { node (id: "%s") { type }}' % id}
+
         r = requests.post(
             urljoin(self.server, "v0/submission/graphql"),
             headers=self.headers,
             data=json.dumps(query),
             verify=self.verify)
+
         if r.status_code == 200:
             result = r.json()
+
             if 'errors' in result:
                 raise Exception("Fail to query file type: {}".format(', '.join(result['errors'])))
+
             nodes = result['data']['node']
             if len(nodes) == 0:
                 raise Exception("File with id {} not found".format(id))
+
             file_type = nodes[0]['type']
+
         else:
             raise Exception(r.text)
+        # </file_type>
+
+        # get metadata about file_type
+        query = {'query': 'query Files { %s (id: "%s") { project_id, file_name }}' % (file_type, id)}
 
-        query = {'query':
-                 """query Files { %s (id: "%s") { project_id, file_name }}""" % (file_type, id)}
         r = requests.post(
             urljoin(self.server, "v0/submission/graphql"),
             headers=self.headers,
             data=json.dumps(query),
             verify=self.verify)
+
         if r.status_code == 200:
+
             result = r.json()
             if 'errors' in result:
                 raise Exception("Fail to query project_id and file_name: {}"
                     .format(', '.join(result['errors'])))
-            for node in result['data'][file_type]:
-                self._metadata = node
+
+            # get first result only
+            if len(result['data'][file_type]) > 0:
+                self._metadata = result['data'][file_type][0]
                 return self._metadata
+
             raise Exception("File with id {} not found".format(id))
+
         else:
             raise Exception("Fail to get filename: {}".format(r.text))
+        # </metadata>
```

Comment:
ah, this is denoting the purpose of the proceeding code block? could this be pulled into a function?